### PR TITLE
Refactor to use security groups rather than IP ranges

### PR DESCRIPTION
**Pull request by niklasvincent**
This change uses security groups rather than IP ranges for access to Mongo nodes, OpsManager and the OpsManager backup node.

The idea is to maintain a separate CloudFormation template with the actual security groups.

### CODE REVIEW

**Comment by philmcmahon** [comment]
This looks good to me. Does this make sense to you @sihil ?

File: cloudformation/mongo-opsmanager-backup.template [hunk 1]
```diff
@@ -346,7 +351,7 @@
             "IpProtocol": "tcp",
             "FromPort": "22",
             "ToPort": "22",
-            "CidrIp": "10.0.0.0/8"
+            "CidrIp": { "Ref": "SSHAccessCIDR" }
```
Comment (sihil) [inline cloudformation/mongo-opsmanager-backup.template:354]: I'd make this smaller. 10.249.0.0/16 would be a good start as that eliminates editorial.

File: cloudformation/mongo-opsmanager-server.template [hunk 2]
```diff
@@ -284,7 +293,7 @@
             "IpProtocol": "tcp",
             "FromPort": "8080",
             "ToPort": "8081",
-            "SourceSecurityGroupId": { "Ref": "BackupLoadBalancerSecurityGroup" }
+            "SourceSecurityGroupId": { "Ref": "BackupLoadBalancerSecurityGroups" }
```
Comment (sihil) [inline cloudformation/mongo-opsmanager-server.template]: Have you tested this with multiple items in the list? What happens when there is more than one group?

**Comment by sihil** [comment]
In principal this is a good change. I'm not convinced that the list of security groups will work (although would like it if it does).

File: cloudformation/mongo-opsmanager-server.template (hunk 2 above)
Comment (niklasvincent) [inline cloudformation/mongo-opsmanager-server.template]: Good point, looking into that now.

File: cloudformation/mongo-opsmanager-backup.template (hunk 1 above)
Comment (niklasvincent) [inline cloudformation/mongo-opsmanager-backup.template:354]: I've made the default for `SSHAccessCIDR` 10.249.0.0/16 across all Mongo related templates.

File: cloudformation/mongo-opsmanager-server.template [hunk 3]
```diff
@@ -284,12 +311,44 @@
...
+        "GroupDescription": "Allow connections between OpsManager MongoDB instances",
+        "VpcId": {
+          "Ref": "VpcId"
+        }
+      }
+    },
+    "OpsManagerReplicationSecurityGroupIngress": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": { "Ref": "ReplicationSecurityGroup" },
```
Comment (philmcmahon) [inline cloudformation/mongo-opsmanager-server.template]: Copy paste error I think - should be OpsManagerReplicationSecurityGroup

File: cloudformation/mongo-opsmanager-server.template [hunk 4]
```diff
@@ -284,12 +311,44 @@
...
+        "IpProtocol": "tcp",
+        "FromPort": "27017",
+        "ToPort": "27018",
+        "SourceSecurityGroupId": { "Ref": "ReplicationSecurityGroup" }
```
Comment (philmcmahon) [inline cloudformation/mongo-opsmanager-server.template]: (as above)

**Comment by niklasvincent** [comment]
We've used the `mongo-opsmanager.template` modifications in CODE for 4 days now and it's worked well. Today I'm going to CloudForm the stack we use for our PROD nodes.

What I might do is reduce the scope of this PR and save the changes to `mongo-opsmanager-server.template` for later. Hopefully then it would be easier to review.

**Comment by niklasvincent** [comment]
I've reduced the scope of this by quite a lot now. Hopefully it should be easier to review.

**Comment by philmcmahon** [comment]
:+1:

**Pull request merged**